spring/src/main/java/io/seata/spring/annotation/datasource/SeataDataSourceBeanPostProcessor.java

```diff
@@ -105,4 +113,62 @@ private Object handleMethodProxy(DataSourceProxy dataSourceProxy, Method method,
             }
         }
     }
+
+    /**
+     * is auto proxied by seata
+     *
+     * @param bean
+     * @return true, if this bean has been auto-proxied by seata
+     */
+    private boolean isAutoProxiedBySeata(Object bean) throws NoSuchFieldException, IllegalAccessException {
+        if (bean instanceof DataSourceProxy) {
+            return true;
+        }
+        //handle Spring AOP
+        Object proxyTargetObject = bean;
+        if (AopUtils.isAopProxy(proxyTargetObject)) {
+            try {
+                proxyTargetObject = SpringProxyUtils.getAdvisedSupport(bean).getTargetSource().getTarget();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+        Field field = null;
+        //handle Normal proxy object
+        if (ClassUtils.isCglibProxy(proxyTargetObject)) {
+            //CGLIB Proxy
+            field = proxyTargetObject.getClass().getDeclaredField("CGLIB$CALLBACK_0");
+        } else if (Proxy.isProxyClass(proxyTargetObject.getClass())) {
+            //JDK Proxy
+            field = proxyTargetObject.getClass().getDeclaredField("h");
+        }
+        return doCheckAutoProxy(field, proxyTargetObject);
+    }
+
+    /**
+     * do check auto proxy
+     *
+     * @param field
+     * @param proxiedObject
+     * @return
+     * @throws IllegalAccessException
+     */
+    private boolean doCheckAutoProxy(Field field, Object proxiedObject) throws IllegalAccessException {
+        if (null == field) {
+            return false;
+        }
+        field.setAccessible(true);
+        Object fieldObject = field.get(proxiedObject);
+        return Stream.of(fieldObject.getClass().getDeclaredFields()).anyMatch(f -> {
+            f.setAccessible(true);
+            Object targetObject;
+            try {
+                targetObject = f.get(fieldObject);
+            } catch (IllegalAccessException e) {
+                throw new RuntimeException(e);
+            }
+            return targetObject instanceof DataSourceProxy;
```

Comment:
In JDK Proxy case, what invocationHandler has a field of type DataSourceProxy?

Reply:
When datasource bean has been proxied once.